zhaquirks/ikea/somrigsmartbtn.py

```diff
@@ -0,0 +1,166 @@
+                ],
+            },
+            # <SimpleDescriptor endpoint=2  profile=260  device_type=6
+            # device_version=1
+            # input_clusters=[0, 3, 4, 64640]
+            # output_clusters=[3, 4, 6, 8, 64640]>
+            2: {
+                PROFILE_ID: zha.PROFILE_ID,
+                DEVICE_TYPE: zha.DeviceType.REMOTE_CONTROL,
+                INPUT_CLUSTERS: [
+                    Basic.cluster_id,
+                    Identify.cluster_id,
+                    Groups.cluster_id,
+                    ShortcutV2Cluster.cluster_id,
+                ],
+                OUTPUT_CLUSTERS: [
+                    Identify.cluster_id,
+                    Groups.cluster_id,
+                    OnOff.cluster_id,
+                    LevelControl.cluster_id,
+                    ShortcutV2Cluster.cluster_id,
+                ],
+            },
+        },
+    }
+
+    replacement = {
+        ENDPOINTS: {
+            1: {
+                PROFILE_ID: zha.PROFILE_ID,
+                DEVICE_TYPE: zha.DeviceType.REMOTE_CONTROL,
+                INPUT_CLUSTERS: [
+                    Basic.cluster_id,
+                    PowerConfig1AAACluster,
+                    Identify.cluster_id,
+                    Groups.cluster_id,
+                    PollControl.cluster_id,
+                    LightLink.cluster_id,
+                    IKEA_CLUSTER_ID,
+                    ShortcutV2Cluster,
+                ],
+                OUTPUT_CLUSTERS: [
+                    Identify.cluster_id,
+                    Groups.cluster_id,
+                    OnOff.cluster_id,
+                    LevelControl.cluster_id,
+                    Ota.cluster_id,
+                    LightLink.cluster_id,
+                    ShortcutV2Cluster,
+                ],
+            },
+            2: {
+                PROFILE_ID: zha.PROFILE_ID,
+                DEVICE_TYPE: zha.DeviceType.REMOTE_CONTROL,
+                INPUT_CLUSTERS: [
+                    Basic.cluster_id,
+                    Identify.cluster_id,
+                    Groups.cluster_id,
+                    ShortcutV2Cluster,
+                ],
+                OUTPUT_CLUSTERS: [
+                    Identify.cluster_id,
+                    Groups.cluster_id,
+                    OnOff.cluster_id,
+                    LevelControl.cluster_id,
+                    ShortcutV2Cluster,
+                ],
+            },
+        },
+    }
+
+    device_automation_triggers = {
+        (PRESSED, BUTTON_1): {ENDPOINT_ID: 1, COMMAND: COMMAND_M_INITIAL_PRESS},
+        (SHORT_PRESS, BUTTON_1): {ENDPOINT_ID: 1, COMMAND: COMMAND_M_SHORT_RELEASE},
+        (DOUBLE_PRESS, BUTTON_1): {
+            ENDPOINT_ID: 1,
+            COMMAND: COMMAND_M_MULTI_PRESS_COMPLETE,
+        },
+        (LONG_PRESS, BUTTON_1): {ENDPOINT_ID: 1, COMMAND: COMMAND_M_LONG_PRESS},
+        (LONG_RELEASE, BUTTON_1): {ENDPOINT_ID: 1, COMMAND: COMMAND_M_LONG_RELEASE},
+
```

Comment:
No blank line in the code block so yo you is getting LINT error.

Reply:
Thanks for reviewing, its my first contribution, and have to admit that I have just copied the Symfonisk2 quirk, and shaved away till it was working well. 

Can you elaborate on your question? I do not understand what you are asking.

Reply:
If all is working OK then its great.
I can testing it later in the week then im back home.

Reply:
Can you help me on how to ad?

created file in: zha_quirkls/somrigsmartbtn.py

Copied your code in that file, and added path in config
    
# automatic OTA update for IKEA Trådfri devices
zha:
   custom_quirks_path: /config/zha_quirks/
   zigpy_config:
    ota:
      ikea_provider: true
     
     
    Its not working. AM i missing something?  Does is require any other settings?

Reply:
Using Home Assistant Supervised On Docker installtion
with instruction from here:
https://github.com/home-assistant/supervised-installer

Reply:
ITS WORS
Didnt copy the code correctly :) need to go to RAW view....

Reply:
THANK YOU

Reply:
It works, but is shomehow slower other ikea remote, takes almost a second for it to trigger in HA

Reply:
I can confirm that it is slow, I thought this was normal for IKEA (the rest of my setup is mainly Sonoff and Hue).

Reply:
Yes, the others are superfast, Expceilly for diming the delay is quite bad...

Reply:
Checking the log, it first fires "Initial Press event was fired with parameters: {'new_position': 0}" and 1 sek later "Short Press - Button 1 event" tried to remove initial Press event in quirk, why does it need the iniitla in the first place? is this just the nature of quirks being rather unresponsive?

Reply:
This device is using ZCL 9 (matter) commands that is not in Zigbbe ZCL but is coming in the next update of it.
Dont try deleting any commands then its braking  the logic that its being used in the future and its not bringing any good.
If like knowing how the logic is working download the later Matter ZCL and find the switch cluster and you is having all there.

Reply:
Thanks for your answer. Ok. I will undelete it.
I am not understanding the rest you were sying 🤪 are you saying it’s possible to reduce that delay?
Appreciate your contribution

Reply:
No by the Matter ZCL its always sending one initial press and must waiting for more presses and then all is don its sending one last press so its not possible as long using this command sets (Aqara and other is not using standard commands so they is not certified Zigbee device).

Reply:
I see :)
So somrig button is not really suitable for  accurate dimming. 
weird the other ikea buttons work just fine, very fast. I hope Ikea will not mess things up even further… maybe they don't want other people using it not in their way?
Unfortunately i couldn't buy older buttons. Somerig where the only ones in stock…

Reply:
**Way** using one Somrig  for dimmer then you can using Rodet and binding it to group / devices and its also working  then you host  system is off line **?**

Its like baying one large SUV and living in old town of Stockholm and you cant driving it and not parking it there =  better with one small Smart that working perfect for it.

Use the right device for the right function or you is only getting problems and Somrig is sending clicks for triggering things in automatons that is one online function and not autonomous Zigbee light commands that Roddret is doing (its one on/off dimmer controller).

Reply:
Yes. I dont have any Rodrets unfortunately . They don't sell them here in Switzerland anymore. Only Somrig.

Reply:
"anymore"  = its only out of stocks then its the standard dimmer switch and shall being in all stores WW in January.